main.py

```diff
@@ -760,16 +778,17 @@ def _send_flow_mods(
                     self._add_flow_mod_sent(
                         flow_mod.header.xid,
                         flow,
-                        build_command_from_flow_mod(flow_mod),
+                        command,
+                        owner
                     )
                 self._send_napp_event(switch, flow, "pending")
         return flow_mods
 
-    def _add_flow_mod_sent(self, xid, flow, command):
+    def _add_flow_mod_sent(self, xid, flow, command, owner):
         """Add the flow mod to the list of flow mods sent."""
         if len(self._flow_mods_sent) >= self._flow_mods_sent_max_size:
             self._flow_mods_sent.popitem(last=False)
-        self._flow_mods_sent[xid] = (flow, command)
+        self._flow_mods_sent[xid] = (flow, command, owner)
```

Comment:
In the future we can refactor this tuple here as a dataclass, probably not worth at the moment since it's unlikely that it'll more than three values here, and a 3-value tuple is still somewhat readable, but let's keep that in mind if we stumble upon this in the future again.